odoo-openmrs/src/main/java/com/ozonehis/eip/odoo/openmrs/processors/ServiceRequestProcessor.java

```diff
@@ -89,35 +89,52 @@ public void process(Exchange exchange) {
-                                    patient.getIdPart(),
-                                    producerTemplate);
+                        if (isActiveStatus || ("u".equals(eventType) && isCompletedStatus)) {
+                            if (saleOrder != null) {
+                                saleOrderHandler.updateSaleOrderIfExistsWithSaleOrderLine(
+                                        serviceRequest,
+                                        saleOrder,
+                                        encounterVisitUuid,
+                                        partner.getPartnerId(),
+                                        patient.getIdPart(),
+                                        producerTemplate);
+                            } else {
+                                saleOrderHandler.createSaleOrderWithSaleOrderLine(
+                                        serviceRequest,
+                                        encounter,
+                                        partner,
+                                        encounterVisitUuid,
+                                        patient.getIdPart(),
+                                        producerTemplate);
+                            }
                         } else {
-                            saleOrderHandler.createSaleOrderWithSaleOrderLine(
-                                    serviceRequest,
-                                    encounter,
-                                    partner,
-                                    encounterVisitUuid,
-                                    patient.getIdPart(),
-                                    producerTemplate);
+                            // Executed when MODIFY option is selected in OpenMRS for other statuses
+                            saleOrderHandler.deleteSaleOrderLine(serviceRequest, encounterVisitUuid, producerTemplate);
                         }
                     } else {
                         // Executed when MODIFY option is selected in OpenMRS
                         saleOrderHandler.deleteSaleOrderLine(serviceRequest, encounterVisitUuid, producerTemplate);
                     }
                 } else if ("d".equals(eventType)) {
-                    // Executed when DISCONTINUE option is selected in OpenMRS
-                    saleOrderHandler.deleteSaleOrderLine(serviceRequest, encounterVisitUuid, producerTemplate);
-                    saleOrderHandler.cancelSaleOrderWhenNoSaleOrderLine(
-                            partner.getPartnerId(), encounterVisitUuid, producerTemplate);
+                    // Executed when a DISCONTINUE option is selected in OpenMRS
+                    // When an Order in OpenMRS is fulfilled, the status is changed to COMPLETED. Therefore, no action
+                    // is taken.
+                    if (serviceRequest.getStatus().equals(ServiceRequest.ServiceRequestStatus.COMPLETED)) {
+                        log.info("ServiceRequest status is COMPLETED. No action taken on Sale Order.");
```

Comment:
Should we log some id as well so that we know which sale order/serviceRequest the log is for.